__tests__/io-test.ts

```diff
@@ -682,14 +725,49 @@ describe('io', () => {
           "0a1e71cf4d0931e81f494d5a73a550016814e15a/53e7a9a0",
           "60a67ad5c653f5b1a6537d9a6aee56c0662c0e35/b9046d34",
           "60a67ad5c653f5b1a6537d9a6aee56c0662c0e35/092271fa",
+          "60a67ad5c653f5b1a6537d9a6aee56c0662c0e35/cc71e5f9",
+        ]
+      `);
+    });
+
+    it('creates items to expire', async () => {
+      const expiredBatches: Map<string, TodoData> = buildTodoData(
+        tmp,
+        getFixture('new-batches', tmp)
+      );
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const expiredTodo: TodoData = expiredBatches.get(
+        '60a67ad5c653f5b1a6537d9a6aee56c0662c0e35/cc71e5f9'
+      )!;
+
+      expiredTodo.errorDate = subDays(getDatePart(), 1).getTime();
+
+      // eslint-disable-next-line unicorn/no-unreadable-array-destructuring
+      const [, , , expired] = await getTodoBatches(
+        buildTodoData(tmp, getFixture('new-batches', tmp)),
+        expiredBatches,
+        { shouldRemove: () => true }
+      );
+
+      expect([...expired.keys()]).toMatchInlineSnapshot(`
+        Array [
+          "60a67ad5c653f5b1a6537d9a6aee56c0662c0e35/cc71e5f9",
         ]
       `);
     });
 
     it('creates all batches', async () => {
-      const [add, remove, stable] = await getTodoBatches(
+      const existingBatches = buildTodoData(tmp, getFixture('existing-batches', tmp));
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Hmm, where is the non-null assertion?

Reply:
nvm, I just misread (thought "next line" meant literally the next line, not the next statement).

Reply:
Ya. I need to fix this as it's indicating our types are slightly misrepresented. I'll do that in a follow up PR.